system/lib/emmalloc.c

```diff
@@ -71,11 +71,10 @@ static_assert((((int32_t)0x80000000U) >> 31) == -1, "This malloc implementation
 static_assert(alignof(max_align_t) == 8, "max_align_t must be correct");
 
 #ifdef EMMALLOC_NO_STD_EXPORTS
-#define EMMALLOC_EXPORT
 #define EMMALLOC_ALIAS(ALIAS, ORIGINAL)
 #else
 #define EMMALLOC_EXPORT __attribute__((weak, __visibility__("default")))
-#define EMMALLOC_ALIAS(ALIAS, ORIGINAL) extern __typeof(ORIGINAL) ALIAS __attribute__((alias(#ORIGINAL)));
+#define EMMALLOC_ALIAS(ALIAS, ORIGINAL) extern __typeof(ORIGINAL) ALIAS __attribute__((weak, alias(#ORIGINAL)));
```

Comment:
Why do we need to make these weak?

Reply:
This is to match the previous behavior.

If we don't do this then e.g. `malloc` would not be weak, so the user couldn't override it.

Reply:
Oh sorry.  I see the existing EMMALLOC_EXPORT is weak... I'm not sure it should be but lgtm given that

Reply:
Also dlmalloc.c uses weak aliases for all of these..